### PR TITLE
portmaster: init at 1.0.0

### LINKED ISSUE #174194: Package Safing's Portmaster
**Project description**
per-application firewall to block trackers etc.

**Metadata**

* homepage URL: https://safing.io/portmaster/
* source URL: https://github.com/safing/portmaster, https://github.com/safing/portmaster-ui
* license: AGPL3, GPL3+
* platforms: Linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #174194

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ncfavier — APPROVED** [review]

File: pkgs/applications/networking/portmaster/default.nix
Comment (ncfavier) [inline pkgs/applications/networking/portmaster/default.nix:3]: Nit: `lib` goes first

**Comment by ncfavier** [comment]
Have you seen the discussion at https://github.com/safing/portmaster/issues/306?

**Comment by urandom2** [comment]
no, that slipped by my radar when I was doing the packaging work; this should not prevent us from merging it into nixpkgs? that would be a blocker for a nixos module.

**Comment by ncfavier** [comment]
I guess so, does this package work correctly on a non-NixOS system?

**Comment by urandom2** [comment]
I have not tested that myself directly, but the build process is fundamentally similar to upstream, and I can get he binary to emit basic information. if you would like can could wait for a user to confirm?

**Comment by ncfavier** [comment]
That would be good. cc @dotlambda @SuperSandro2000

**Comment by urandom2** [comment]
hopefully the original submitter sees this, otherwise I will concert it to a draft to reflect that additional review is required before merge.

**Review by dotlambda — COMMENTED** [review]
Can we disable automatic updates?

File: pkgs/applications/networking/portmaster/default.nix
Comment (urandom2) [reply pkgs/applications/networking/portmaster/default.nix:20]: integration tests require root access

**Comment by urandom2** [comment]
I trust it is technically possible, but I am not familiar enough with portmaster to know how to do it

**Comment by SuperSandro2000** [comment]
Please don't post PRs every week on discourse.

> I guess so, does this package work correctly on a non-NixOS system?

I doubt that the package just works on NixOS. It needs to be tested and if it broken we should wrap it in an fhs or come up with something different.

**Comment by urandom2** [comment]
> Please don't post PRs every week on discourse.

there is not any official usage information on the discourse thread; do we need to add some? the latest information I was given was to wait a week before re-requesting a review on discourse. if this is not the case, please let me know how to use that thread.

my only intention was to signal that "this pr is ready for outside review", and simply resolving threads or pushing a new commit does not seem to be enough to get another review.

> I doubt that the package just works on NixOS. It needs to be tested and if it broken we should wrap it in an fhs or come up with something different.

so, I am not a user of this tool, and thus cannot fully exercise all the functionality. what is the correct burden of proof for packaging?

that said, since there are some clear issues called out in safing/portmaster#306 and seems pretty active, I am happy to step back so one of them can drive this forward. this change is formally donated to anybody who wants to pick it up.

**Comment by SuperSandro2000** [comment]
> there is not any official usage information on the discourse thread; do we need to add some? the latest information I was given was to wait a week before re-requesting a review on discourse. if this is not the case, please let me know how to use that thread.

We don't need bureaucratize that, just don't post all your PRs every week.
Pushing new commits is enough to trigger notifications but for me personally the notification list is currently just around 1000, so things slip by. Resolving conversations does not send any notifications.





> so, I am not a user of this tool, and thus cannot fully exercise all the functionality. what is the correct burden of proof for packaging?

At least basic functionality should work before a PR is created unless stated somewhere visible.

> that said, since there are some clear issues called out in [safing/portmaster#306](https://github.com/safing/portmaster/issues/306) and seems pretty active, I am happy to step back so one of them can drive this forward. this change is formally donated to anybody who wants to pick it up.

FYI: all your contributions are licensed under MIT already anyway.

**Comment by Nyabinary** [comment]
@urandom2 Hmm can we bring this back?

**Comment by Nyabinary** [comment]
Bumping this again lol

**Pull request closed without merging**
